app/assets/src/components/visualizations/Histogram.js

```diff
@@ -274,10 +276,17 @@ export default class Histogram {
     if (!this.data) return;
 
     let colors = this.getColors();
-
+    // cannot pass in 0 if using log scale, since log(0) is -Infinity,
+    // so increment everything by 1
+    if (this.options.xScaleLog) {
+      for (let i = 0; i < this.data.length; i++) {
+        this.data[i] = this.data[i].map(d => d + 1);
+      }
+    }
     const domain = this.getDomain();
 
-    let x = scaleLinear()
+    let x = this.options.xScaleLog ? scaleLog() : scaleLinear();
+    x
```

Comment:
Can you use the clamp function as shown here?
https://stackoverflow.com/questions/11322651/how-to-avoid-log-zero-in-graph-using-d3-js
You can clamp the domain to [1, max(maxs)]

This way, you can remove all the special cases around adding +1 for log.

Reply:
Doesn't this mean the bin around 0 will be dropped entirely? Usually that bin contains the most information and dropping it would be very misleading